app/src/main/java/com/victor/adasfeed/ProfileActivity.kt

```diff
@@ -45,19 +65,77 @@ class ProfileActivity : AppCompatActivity() {
             }
 
             user?.let { safeUser ->
-                textUserName.text = safeUser.userName
-                textNickname.text = safeUser.userNickname
-                imageUser.setImageResource(safeUser.imageUser)
-                uri = safeUser.tel?.let {
-                    Uri.parse("tel:${safeUser.tel}")
+                val userName = safeUser.userName
+                val userNickname = safeUser.userNickname
+                val imageUser1 = safeUser.imageUser
+                val tel = safeUser.tel
+                userProfile = User(userName, userNickname,imageUser1,tel)
+                textUserName.text = userName
+                textNickname.text = userNickname
+                imageUser.setImageResource(imageUser1)
+                etUsername.setText(userName)
+                etNickname.setText(userNickname)
+
+                uri1 = tel?.let {
+                    Uri.parse("tel:$tel")
                 } ?: Uri.parse("tel:")
             }
         }
+        return uri1
+    }
 
+    private fun setListeners(uri: Uri?) {
         buttonCall.setOnClickListener {
             val intent = Intent(Intent.ACTION_DIAL, uri)
             startActivity(intent)
         }
+
+        btnReturnToFeed.setOnClickListener {
+            val result = Intent().putExtra("Updated user", userProfile)
+            setResult(Activity.RESULT_OK, result)
+            finish()
+        }
+
+        fabSaveButton.setOnClickListener {
+            saveDataUserProfile(it)
+        }
+
+        etUsername.addTextChangedListener { editable ->
+            textUserName.text = editable.toString()
+        }
+        etNickname.addTextChangedListener { editable ->
+            textNickname.text =
+                editable.toString().ifBlank {
+                    "@${editable.toString()}"
+                }
+        }
+    }
+
+    private fun saveDataUserProfile(view: View) {
+        if (isValidForms()){
+            updateProfile()
+            showSnackbarWithText("Perfil atualizado",view)
+        } else {
+            showSnackbarWithText("Username ou nickname não podem estar vazios",view)
+        }
+    }
+
+    private fun showSnackbarWithText(text: String,view: View) {
+        val snackbar = Snackbar.make(view, text, Snackbar.LENGTH_LONG)
+        snackbar.setAnchorView(R.id.buttonReturnToFeed)
+        snackbar.show()
+    }
+
+    private fun isValidForms(): Boolean = when{
+        etUsername.text.toString().isNotBlank() -> true
+        etNickname.text.toString().isNotBlank() -> true
+        else -> false
+    }
+
+    private fun updateProfile() {
+        val username = etUsername.text.toString()
+        val nickname = etNickname.text.toString()
```

Comment:
Could use extension function:
`private fun String?.orDefaultValue(default: String) = if (this.isNullOrBlank() default else this`